### PR TITLE
HBASE-24446 Use EnvironmentEdgeManager to compute clock skew in Master

### PR DESCRIPTION
**Pull request by sguggilam**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |  13m 46s |  Docker failed to build yetus/hbase:7cf97016b9.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/1835 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/1/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m  0s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 44s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   0m 49s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   1m 54s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 22s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 43s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   3m 14s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 10s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 48s |  hbase-server: The patch generated 1 new + 40 unchanged - 1 fixed = 41 total (was 41)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 10s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   3m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 152m 20s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   | 197m 52s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1835 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 9ca971d2f37f 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1835/out/precommit/personality/provided.sh |
| git revision | branch-1 / d889c7b |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/2/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/2/testReport/ |
| Max. process+thread count | 3822 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by sguggilam** [comment]
@virajjasani  This is change for branch-1 . Please review

**Comment by sguggilam** [comment]
@apurtell

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java [hunk 1]
```diff
@@ -78,13 +75,18 @@
...
 import org.apache.hadoop.hbase.util.RetryCounterFactory;
 import org.apache.hadoop.hbase.util.Triple;
 import org.apache.hadoop.hbase.zookeeper.ZKUtil;
 import org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher;
 import org.apache.zookeeper.KeeperException;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.ServiceException;
+
```
Comment (virajjasani) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java:89]: @sguggilam looks like import order is incorrect as per checkstyle error:
```
ServerManager.java:86:import com.google.common.annotations.VisibleForTesting;: Wrong order for 'com.google.common.annotations.VisibleForTesting' import. [ImportOrder]
```
Maybe you can put these import in their original place, (could be IDE settings issue)

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 51s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 39s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 40s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   2m 59s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 56s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 40s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   0m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 33s |  hbase-server: The patch generated 0 new + 40 unchanged - 1 fixed = 40 total (was 41)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 33s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   2m 49s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 125m 30s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 166m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.11 Server=19.03.11 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1835 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 225ed691a254 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1835/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4118383 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/3/testReport/ |
| Max. process+thread count | 4295 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1835/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java (hunk 1 above)
Comment (sguggilam) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java]: It was using default alphabetical sorting order....imported the hbase import file in eclipse now...fixed and pushed the change

**Comment by apurtell** [comment]
Belated +1

**Pull request merged**